module/documents/activity/order.mjs

```diff
@@ -0,0 +1,278 @@
+import ActivityMixin from "./mixin.mjs";
+import OrderActivityData from "../../data/activity/order-data.mjs";
+import OrderUsageDialog from "../../applications/activity/order-usage-dialog.mjs";
+
+/**
+ * @typedef {ActivityUseConfiguration} OrderUseConfiguration
+ * @property {object} [costs]
+ * @property {number} [costs.days]               The cost of executing the order, in days.
+ * @property {number} [costs.gold]               The cost of executing the order, in gold.
+ * @property {boolean} [costs.paid]              Whether the gold cost has been paid.
+ * @property {object} [craft]
+ * @property {string} [craft.item]               The item being crafted or harvested.
+ * @property {number} [craft.quantity]           The quantity of items to harvest.
+ * @property {object} [trade]
+ * @property {boolean} [trade.sell]              Whether the trade was a sell operation.
+ * @property {object} [trade.stock]
+ * @property {boolean} [trade.stock.stocked]     Whether the order was to fully stock the inventory.
+ * @property {boolean} [trade.stock.value]       The base value of goods transacted.
+ * @property {object} [trade.creatures]
+ * @property {string[]} [trade.creatures.buy]    Additional animals purchased.
+ * @property {boolean[]} [trade.creatures.sell]  Whether a creature in a given slot was sold.
+ * @property {number} [trade.creatures.price]    The base value of the animals sold.
+ */
+
+/**
+ * An activity for issuing an order to a facility.
+ */
+export default class OrderActivity extends ActivityMixin(OrderActivityData) {
+  /* -------------------------------------------- */
+  /*  Model Configuration                         */
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  static metadata = Object.freeze(foundry.utils.mergeObject(super.metadata, {
+    configurable: false,
+    type: "order",
+    img: "systems/dnd5e/icons/svg/activity/order.svg",
+    title: "DND5E.FACILITY.Order.Issue",
+    usage: {
+      actions: {
+        pay: OrderActivity.#onPayOrder
+      },
+      chatCard: "systems/dnd5e/templates/chat/order-activity-card.hbs",
+      dialog: OrderUsageDialog
+    }
+  }, { inplace: false }));
+
+  /* -------------------------------------------- */
+  /*  Activation                                  */
+  /* -------------------------------------------- */
+
+  /** @override */
+  shouldHideChatButton(button, message) {
+    return !this.actor.isOwner;
```

Comment:
`ChatMessage#_displayChatActionButtons` should already be hiding all buttons from non-owners/non-GMs.

Reply:
I was sure I tested it, and the buttons were still visible to a user that did not own the Actor. I will leave it for now, but we can remove it if we do some testing and determine it's definitely redundant.